### PR TITLE
Add SKL and ICL as architectures

### PR DESCRIPTION
**Pull request by masterleinad**
In response to https://github.com/kokkos/kokkos/pull/4929#issuecomment-1123404821.
I also removed linking with `rtm` since transactional memory is not used anymore anyway.

### CODE REVIEW

File: cmake/kokkos_arch.cmake [hunk 1]
```diff
@@ -53,7 +53,9 @@ DECLARE_AND_CHECK_HOST_ARCH(WSM               "Intel Westmere CPU")
 DECLARE_AND_CHECK_HOST_ARCH(SNB               "Intel Sandy/Ivy Bridge CPUs")
 DECLARE_AND_CHECK_HOST_ARCH(HSW               "Intel Haswell CPUs")
 DECLARE_AND_CHECK_HOST_ARCH(BDW               "Intel Broadwell Xeon E-class CPUs")
+DECLARE_AND_CHECK_HOST_ARCH(ICL               "Intel Ice Lake Client CPUs (AVX512)")
 DECLARE_AND_CHECK_HOST_ARCH(ICX               "Intel Ice Lake Server CPUs (AVX512)")
+DECLARE_AND_CHECK_HOST_ARCH(SKL               "Intel Sky Lake Xeon E-class Client CPUs")
```
[the remark this answers is not in the document]
Comment (masterleinad) [reply cmake/kokkos_arch.cmake]: Done.

[the remark this answers is not in the document]
Comment (masterleinad) [reply cmake/kokkos_arch.cmake]: It's `Ice Lake` and `Skylake`.

**Review by crtrott — APPROVED** [review]

File: cmake/kokkos_arch.cmake [hunk 2]
```diff
@@ -53,8 +53,10 @@ DECLARE_AND_CHECK_HOST_ARCH(WSM               "Intel Westmere CPU")
 DECLARE_AND_CHECK_HOST_ARCH(SNB               "Intel Sandy/Ivy Bridge CPUs")
 DECLARE_AND_CHECK_HOST_ARCH(HSW               "Intel Haswell CPUs")
 DECLARE_AND_CHECK_HOST_ARCH(BDW               "Intel Broadwell Xeon E-class CPUs")
-DECLARE_AND_CHECK_HOST_ARCH(ICX               "Intel Ice Lake Server CPUs (AVX512)")
-DECLARE_AND_CHECK_HOST_ARCH(SKX               "Intel Sky Lake Xeon E-class HPC CPUs (AVX512)")
+DECLARE_AND_CHECK_HOST_ARCH(ICL               "Intel Ice Lake Client CPUs (AVX512)")
+DECLARE_AND_CHECK_HOST_ARCH(ICX               "Intel Ice Lake Xeon Server CPUs (AVX512)")
+DECLARE_AND_CHECK_HOST_ARCH(SKL               "Intel Skylake Client CPUs")
```
Comment (dalg24) [inline cmake/kokkos_arch.cmake:58]: Is this one not missing the `(AVX512)`?

Comment (masterleinad) [reply cmake/kokkos_arch.cmake:58]: No, not all of them support AVX512.

**Pull request merged**